test/basictests.jl

```diff
@@ -9,37 +9,52 @@ const Dual64 = ForwardDiff.Dual{Nothing, Float64, 1}
 n = 8
 A = Matrix(I, n, n)
 b = ones(n)
+# Real-valued systems
 A1 = A / 1;
 b1 = rand(n);
 x1 = zero(b);
+# A2 is similar to A1; created to test cache reuse
 A2 = A / 2;
 b2 = rand(n);
 x2 = zero(b);
+# Complex systems + mismatch types with eltype(tol)
+A3 = (A .+ rand(n) .* im) .|> ComplexF32;
+b3 = rand(n) .|> ComplexF32;
+x3 = zero(b) .|> ComplexF32;
+# A4 is similar to A3; created to test cache reuse
+A4 = (A .+ rand(n) .* im) .|> ComplexF32;
+b4 = b3 / 4 .|> ComplexF32;
+x4 = zero(b) .|> ComplexF32;
```

Comment:
@ChrisRackauckas A bunch of tests failed due to the type mismatch between A3 and A4 (used to be ComplexF64 and ComplexF32), so I made this change; but not sure if this was supposed to be handled internally, or it was expected that the cache mechanism should only work on systems with the same dtype.

Reply:
The cache mechanism only works with the same dtype.